src/libraries/Microsoft.Extensions.Options/gen/Parser.cs

```diff
@@ -439,7 +439,8 @@ private List<ValidatedMember> GetMembersToValidate(ITypeSymbol modelType, bool s
                     // pop the stack
                     _ = _visitedModelTypes.Remove(enumeratedType.WithNullableAnnotation(NullableAnnotation.None));
                 }
-                else if (ConvertTo(attributeType, _symbolHolder.ValidationAttributeSymbol))
+                else if (ConvertTo(attributeType, _symbolHolder.ValidationAttributeSymbol) &&
+                        _compilation.IsSymbolAccessibleWithin(attributeType, _compilation.Assembly))
```

Comment:
Nit: it might make sense to make your `within:` parameter more specific, e.g. pass the symbol of type where the code is being generated. For example the attribute type could private within the type in which code is being generated (and therefore still accessible).

Reply:
Although I am not seeing a way to produce such a problem, I scoped the accessibility check to the validator type anyway. This is safer and can have better perf.